### DISCUSSION TITLE
Using react-instantsearch-nextjs, how to change address url?

### DISCUSSION THREAD

**Discussion by web-programmer-here** [algolia/instantsearch#5929] [Q&A]
I am using nextjs app router and came across this package `react-instantsearch-nextjs` 

You can pass routing as a boolean to enable routing: 
```
<InstantSearchNext searchClient={client} indexName="instant_search" routing>
```

but url is updated like this: 

```
http://localhost:3000/?my_index_name[query]=my_search_term
http://localhost:3000/?my_index_name[page]=2
```

how can I customize these?
so I want 
```
http://localhost:3000/?my_index_name[query]=my_search_term
to be: 
http://localhost:3000/searchterm=my_search_term

http://localhost:3000/?my_index_name[page]=2
to be 
http://localhost:3000/page/2
```

**Comment by web-programmer-here** [comment]
also the docs here: https://www.algolia.com/doc/guides/building-search-ui/going-further/server-side-rendering/react/#enabling-routing says 

> To customize the way InstantSearch maps the state to the route, pass an object to routing, where router has the same options as [history](https://www.algolia.com/doc/api-reference/widgets/history-router/react/), and stateMapping has the same options as the one in [<InstantSearch>](https://www.algolia.com/doc/api-reference/widgets/instantsearch/react/).

but looks like it is not supported by `react-instantsearch-nextjs` yet?

**Comment by Haroenv** [comment]
the arguments are:

```
<InstantSearchNext routing={{ router: { createURL, parseURL ... }, stateMapping }}
```

You don't create the historyRouter yourself

**Comment by web-programmer-here** [comment]
@Haroenv thanks, is there an example or can you please provide a basic example?
I am not sure how to create the url with the structure I want.
I was looking at this: https://www.algolia.com/doc/api-reference/widgets/history-router/react/
